gulpfile.ts

```diff
@@ -0,0 +1,159 @@
+import gulp from 'gulp';
+// eslint-disable-next-line @typescript-eslint/no-var-requires
+const clean = require('gulp-clean');
+import merge from 'merge-stream';
+import ts from 'gulp-typescript';
+import sourcemaps from 'gulp-sourcemaps';
+import terser from 'gulp-terser';
+import uglify from 'gulp-uglify';
+// eslint-disable-next-line @typescript-eslint/no-var-requires
+const rollup = require('gulp-rollup-2').rollup;
+import rollupTypescript from 'rollup-plugin-typescript2';
+
+import minifyPrivatesTransformer from 'ts-transformer-minify-privates';
+import tsMacroTransformer from 'ts-macros';
+import pathsTransformer from 'ts-transform-paths';
+import { CustomTransformerFactory, Program } from 'typescript';
+
+function createProject(overrideSettings?: ts.Settings) {
+  return ts.createProject('tsconfig.json', {
+    getCustomTransformers: (program?: Program) => {
+      if (program === undefined) throw new Error('Program is undefined');
+      return {
+        ...pathsTransformer(program),
+        before: [
+          tsMacroTransformer(program) as unknown as CustomTransformerFactory,
+          minifyPrivatesTransformer(program)
+        ]
+      };
+    },
+    ...overrideSettings
+  });
+}
+
+function terserStream() {
+  return terser({
+    compress: false,
+    output: {
+      beautify: true
+    },
+    mangle: {
+      keep_fnames: true,
+      keep_classnames: true,
+      properties: {
+        regex: /^_private_/
+      }
+    },
+    nameCache: {}
+  });
+}
+
+function rollupStream(outFileName: string): NodeJS.ReadWriteStream {
+  return rollup({
+    input: 'src/index.ts',
+    output: {
+      file: outFileName,
+      format: 'umd',
+      name: 'sdsl'
+    },
+    plugins: [
+      rollupTypescript({
+        tsconfig: 'tsconfig.json',
```

Comment:
You can remove this config because the default value is `tsconfig.json`.

Reply:
ok. I'll remove it right away